### DISCUSSION TITLE
nvim-treesitter[xml]: Error during compilation

### DISCUSSION THREAD

**Discussion by YueXingLiao** [LazyVim/LazyVim#2761] [Q&A] [answered]
I'm a newbie and every time I open my neovim, the message window always shows nvim-treesitter[xml]: Error during compilation, error message.
[nvim-treesitter] [0/1] Download tree-sitter-xml...
[nvim-treesitter] [0/1] Checking out locked revisions
[nvim-treesitter] [0/1] Compiling...
nvim-treesitter [xml]: error during compilation
src/scanner.c: in function 'string_push':
src/scanner.c:28:24: warning: implicit declaration of function 'max' [-Wimplicit-function-declaration] (implicit function declaration)
   28 | uint32_t cap = max(STRING_CAP, string->size << 1);
      | ^~~

collect2.exe: error: ld returned 1 exit status

Translated with DeepL.com (free version)

**Comment by dpetka2001** [comment, marked as answer]
If you're on Windows, you'd better read [this](https://github.com/nvim-treesitter/nvim-treesitter/wiki/Windows-support) and check the requirements that are needed in order for the parsers to be able to be compiled.

**Reply by YueXingLiao** [reply]
I did notice these when configuring LazyVim

Neovim >= 0.9.0 (needs to be built with LuaJIT)
Git >= 2.19.0 (for partial clones support)
a Nerd Font (optional)
a C compiler for nvim-treesitter. see here

The prerequisites are the installation of a compliant C compiler in a Windows environment. At first I thought I just needed to point to the relevant set of commands in an environment variable, but that didn't work until I noticed this:
Make clang the compiler to be used by nvim-treesitter. Put the following line somewhere in your lua config.

require 'nvim-treesitter.install'.compilers = { "clang" } 

Of course, this assumes that LLVM is installed in a Windows environment, and now it looks fine!

Thanks for your help~!

**Reply by JosephBrooksbank** [reply]
FWIW, this appears to also happen when using minGW from choco, ONLY for `xml`. I fixed this by just removing xml, but after seeing this thread I installed `llvm` alongide minGW and tried re-enabling `xml` treesitting, and it works now.
